plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/ConfigSetup.java

```diff
@@ -0,0 +1,228 @@
+package io.jenkins.tools.pluginmanager.cli;
+
+import io.jenkins.tools.pluginmanager.config.Config;
+import io.jenkins.tools.pluginmanager.config.Settings;
+import io.jenkins.tools.pluginmanager.impl.Plugin;
+import java.io.FileNotFoundException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import org.apache.commons.lang3.StringUtils;
+
+
+public class ConfigSetup {
```

Comment:
I think we could likely get rid of this class in favour of the CliOptions class

Reply:
So I'm definitely open to moving things around, but this is the way I've been thinking about it. The CliOptions class is only for getting the options that were set in the Cli.  The ConfigSetup takes what was set in the Cli and creates a configuration class from it.  The Config class is what is actually gets passed to the configuration file.  A lot of the methods are named similarly though.  So you think we should move the ConfigSetup into the CLIOptions class?

Reply:
I wonder about this:
CLIOptions creating a config class
A ConfigBuilder (builder pattern) being created with all defaults there

It could look something like 
```
Config CLIOptions#getConfig() {
  ConfigBuilder builder = Config.builder()
  if (jenkinsUc != null) builder.withJenkinsUc(jenkinsUc)
  ...
  return builder.build();
}
```

it's a bit clunky because you need null checks though, could use Optional#ifPresent...
Or skip the builder if it doesn't give many benefits

Reply:
Ok. Are you cool with me merging this as is and then I can create a Jira task for doing that in a separate PR?

Reply:
Yea can be separate, I'm having a look at it 👍